props/saber_fett263_buttons.h

```diff
@@ -285,6 +445,217 @@ EFFECT(faston); // for EFFECT_FAST_ON
+};
+
+#ifdef FETT263_SAVE_CHOREOGRAPHY
+// Rehearsal / Choreography
+class SavedRehearsal : public ConfigFile {
+public:
+  virtual ReadStatus Read(FileReader* f) override {
+    int file;
+    if (!f || !f->IsOpen()) return ReadStatus::READ_FAIL;
+    memset(clash_rec, 0, sizeof(clash_rec));
+    STDOUT.println("Reading choreo.ini");
+    for (size_t i = 0; i < NELEM(clash_rec); i++) {
+      char action[33];
+      f->skipwhite();
+      f->readVariable(action);
+      if (!strcmp(action, "clash")) {
+        if (SFX_clsh) {
+           file = SFX_clsh.get_min_file();
+        } else {
+           file = SFX_clash.get_min_file();
+        }
+        clash_rec[i].stance = STANCE_CLASH;
+        f->skipspace();
+        clash_rec[i].sound_number1 = (int)(f->readFloatValue() - file);
+        f->skipspace();
+        continue;
+      }
+      if (!strcmp(action, "lockup")) {
+        if (SFX_lock) {
+          file = SFX_lock.get_min_file();
+        } else {
+          file = SFX_lockup.get_min_file();
+        }
+        clash_rec[i].stance = STANCE_LOCKUP;
+        f->skipspace();
+        clash_rec[i].sound_number1 = (int)(f->readFloatValue() - SFX_bgnlock.get_min_file());
+        f->skipspace();
+        clash_rec[i].sound_number2 = (int)(f->readFloatValue() - file);
+        f->skipspace();
+        clash_rec[i].sound_number3 = (int)(f->readFloatValue() - SFX_endlock.get_min_file());
+        f->skipspace();
+        continue;
+      }
+      clash_rec[i].stance = STANCE_END;
+      return ReadStatus:: READ_END;
+      break;
+    }
+  }
+
+  void SetVariable(const char* variable, float v) override {
+  }
+
+  enum FormStance {
+    STANCE_END = 0,
+    STANCE_CLASH,
+    STANCE_LOCKUP,
+  };
+
+  struct FormEntry {
+    FormStance stance;
+    int sound_number1;
+    int sound_number2;
+    int sound_number3;
+  };
+
+  FormEntry clash_rec[32];
+};
+#endif
+
+#ifdef FETT263_EDIT_MODE_MENU
+// Edit Length 
+int length_edit_length = 0;
+
+// Edit Length Preview for Edit Mode
+// LengthEdit uses blade color and creates single white pixel at last pixel, black above last pixel
+template<class BASE, class LIGHTUP, class BLACK = BLACK>
+class LengthEdit {
+public:
+  void run(BladeBase* blade) {
+    base_.run(blade);
+    lightup_.run(blade);
+    black_.run(blade);
+    led_ = length_edit_length;
+  }
+
+  OverDriveColor getColor(int led) {
+    if (led == led_) return lightup_.getColor(led);
+    if (led > led_) return black_.getColor(led);
+    return base_.getColor(led);
+  }
+private:
+  int led_;
+  BASE base_;
+  LIGHTUP lightup_;
+  BLACK black_;
+};
+
+// Edit Style Settings
+
+class IntEdit {
+  public:
+    void run(BladeBase* blade) {  }
+    int getInteger(int led) { return int_edit_; }
+    static void SetIntValue(int value) { int_edit_ = value; }
+ private:
+   static int int_edit_;
+};
+
+int IntEdit::int_edit_ = 0;
+#endif
+  
+// Color List
+static constexpr Color16 color_list_[] = {
+  { 65535, 0, 0 }, // Red
```

Comment:
Red::color(),

Reply:
was not able to get this working, getting compile error on attempt:

c:\users\fernan~1\appdata\local\temp\arduino_build_940745\sketch\props\saber_fett263_buttons.h:631:14: error: call to non-'constexpr' function 'static Color16 Rgb<R, G, B>::color() [with int R = 255; int G = 0; int B = 0]'
  631 |   Red::color(),
      |              ^

I tried adding constexpr to /styles/rgb.h but it creates even more errors so not sure how to proceed

Reply:
updated